src/main.rs

```diff
@@ -232,6 +232,31 @@ fn print_table<T: std::fmt::Display + Clone>(mut input: Vec<T>, columns: usize,
     table.printstd();
 }
 
+fn dep_wanted(preconditions: &When, settings: &serde_json::Value) -> bool {
+    let false_sentinel = Value::Bool(false);
+
+    let has_a_prereq = if !preconditions.all_of.is_empty() {
+        preconditions.all_of.iter().all(|feature| {
+            let has_feature = settings.get(feature).unwrap_or(&false_sentinel);
+            has_feature.as_bool().unwrap_or(false)
+        })
+    } else if !preconditions.any_of.is_empty() {
+        preconditions.any_of.iter().any(|feature| {
+            let has_feature = settings.get(feature).unwrap_or(&false_sentinel);
+            has_feature.as_bool().unwrap_or(false)
+        })
+    } else {
+        false
+    };
+
+    let has_no_exclusions = !preconditions.none_of.iter().any(|feature| {
+        let has_feature = settings.get(feature).unwrap_or(&false_sentinel);
+        has_feature.as_bool().unwrap_or(false)
+    });
+
+    has_a_prereq && has_no_exclusions
+}
```

Comment:
It strikes me that extracting `When` and our directory renderer into a crate would be useful.